<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Introduction into bioinformatics tools - A primer on phylogeny</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Introduction into bioinformatics tools</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./sanger.html" rel="" target="">
 <span class="menu-text">Sanger Sequencing Theory and Data</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./alignment.html" rel="" target="">
 <span class="menu-text">Sequence alignment</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./phylogenetics.html" rel="" target="" aria-current="page">
 <span class="menu-text">A primer on phylogeny</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#phylogenetics" id="toc-phylogenetics" class="nav-link active" data-scroll-target="#phylogenetics">Phylogenetics</a></li>
  <li><a href="#influenza-practical-session" id="toc-influenza-practical-session" class="nav-link" data-scroll-target="#influenza-practical-session">Influenza: Practical session</a></li>
  <li><a href="#practical" id="toc-practical" class="nav-link" data-scroll-target="#practical">Practical</a></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods">Methods:</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">A primer on phylogeny</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Before we get to phylogenetics let’s look at some iris data. As with many other bioinformatic techniques, phylogenetics stems from older methods. Hierarchical clustering is a form of unsupervised machine learning and unlike some of the other methods of this kind, you do not have to specify the number of clusters you want prior to the analysis. The picture below illustrates what we will be looking at.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="media/iris-machinelearning.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">iris</figcaption>
</figure>
</div>
<p>The ‘iris’ dataset is quite famous and is used in many books and tutorials on machine learning. https://en.wikipedia.org/wiki/Iris_flower_data_set</p>
<p>The data set consists of four different attributes, namely: sepal length, sepal width, petal length and petal width (all measured in cm). There are 50 observations for each class and they are like in the image above, versicolor, setosa and virginica.</p>
<p>I could show you the data, but let’s rather look at some fun plots.</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggplot2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'GGally':
  method from   
  +.gg   ggplot2</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>  Sepal.Length Sepal.Width Petal.Length Petal.Width Species
1          5.1         3.5          1.4         0.2  setosa
2          4.9         3.0          1.4         0.2  setosa</code></pre>
</div>
<div class="cell-output-display">
<p><img src="phylogenetics_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The pairs plotted above plot each attribute or variable against each. Why would this be useful?</p>
<p>From the above plot, it is now clear that petal length and width are the attributes which are most discriminating between the 3 different flower species. So if we would like to group these into clusters based on their attributes the above-mentioned would do the best job. Also, remember that the attributes are measured in cm which means that we can use this numeric data to construct a distance matrix. Once we have a distance matrix we can create a dendrogram. So how would we calculate the distance between observations? Remember we have 150 observations, 50 of each specie. We would like to calculate the distance of each observation against each. If you now think about that, if we plot these distances, they should form clusters.</p>
<p>How do we calculate distances? Remember Pythagoras’s theorem?</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="media/pyth.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Pythagoras</figcaption>
</figure>
</div>
<p>Or do you remember getting the distance between two points on a cartesian coordinate system, the image below should help.</p>
<p><img src="media/Euclidean_distance-300x240.png" class="img-fluid" alt="cartesian"> So how do we calculate the distance (d) between points (x1, y1) and (x2,y2)?</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="media/euclidean_eq.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">euclidean</figcaption>
</figure>
</div>
<div class="cell">
<div class="cell-output-display">
<p><img src="phylogenetics_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output-display">
<p><img src="phylogenetics_files/figure-html/unnamed-chunk-2-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>We can see from the above image that we mostly managed to correctly cluster the iris species.</p>
<section id="phylogenetics" class="level2">
<h2 class="anchored" data-anchor-id="phylogenetics">Phylogenetics</h2>
<p>I am borrowing HEAVILY from the link below and I would encourage you to have a look yourself.</p>
<p><a href="https://www.ebi.ac.uk/training/online/course/introduction-phylogenetics">https://www.ebi.ac.uk/training/online/course/introduction-phylogenetics</a></p>
<p>You can also go over to EMBL-EBI Training website <a href="https://www.ebi.ac.uk/training/">https://www.ebi.ac.uk/training/</a> for other free training opportunities as well. Maybe have a look at this introduction video</p>
<iframe width="775" height="299" src="https://www.youtube.com/embed/Rwrw28hQMfE" title="EMBL-EBI: the home of big data in biology" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen="">
</iframe>
<section id="what-is-phylogenetics" class="level4">
<h4 class="anchored" data-anchor-id="what-is-phylogenetics">… what is phylogenetics</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="media/phylo1.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">What is phylogenetics</figcaption>
</figure>
</div>
<p>If you recall from the previous installment of the course we looked at multiple sequence alignment and distance matrices. Like with our flowers in the above sections, we can use a multiple sequence alignment to calculate distance matrices and infer phylogenetic relationships.</p>
<p>Typically phylogeneticists study one of the following types of question: • What are the evolutionary relationships or histories among my species/individuals/genes of interest? • How do sequences evolve? • Can I better describe processes of sequence evolution with a mathematical model?</p>
</section>
<section id="why-phylogenetics" class="level4">
<h4 class="anchored" data-anchor-id="why-phylogenetics">… Why phylogenetics</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="media/phylogenetics1-530x510.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">why phylogenetics</figcaption>
</figure>
</div>
<p><a href="https://www.ebi.ac.uk/training/online/course/introduction-phylogenetics/why-phylogenetics-important">https://www.ebi.ac.uk/training/online/course/introduction-phylogenetics/why-phylogenetics-important</a></p>
<p>As we can see from the images below, a ‘tree of life’ is an old idea long before the discovery of DNA in 1950 and the first DNA sequence in 1970.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="media/old_trees.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Ernst Haeckel and Charles Darwin</figcaption>
</figure>
</div>
<p>The field of phylogenetic analysis is a broad one with various methods and models. Most researchers look at methods and models already used by those before them and learn what they need to know on a need-to-know basis. Also if we would like to look at this topic in depth, I would rather call in an expert to present it. A nice book on the topic is: The Phylogenetic Handbook: A Practical Approach to Phylogenetic Analysis and Hypothesis Testing. ISBN-13: 978-0521730716</p>
<p>Let’s rather apply this technique to solving a pending real word disaster.</p>
</section>
</section>
<section id="influenza-practical-session" class="level2">
<h2 class="anchored" data-anchor-id="influenza-practical-session">Influenza: Practical session</h2>
<p>Influenza A virus is a negative sense, single-stranded, segmented RNA virus. This Flu virus evolves through 2 mechanisms.</p>
<p>Genetic Drift: This is the small changes in the genome where mutations occur within the segments.</p>
<p>Genetic Shift: This is also sometimes called re-assortment and is typical of segmented viruses like influenza A, rotavirus, Crimean-Congo haemorrhagic fever virus and probably any segmented virus.</p>
<p>The figure below illustrates the re-assortment potential of a zoonotic virus. <a href="https://link.springer.com/article/10.1007/s11908-018-0642-9">paper</a></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1007%2Fs11908-018-0642-9/MediaObjects/11908_2018_642_Fig1_HTML.png?as=webp" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Goneau et al.&nbsp;2018</figcaption>
</figure>
</div>
<p>The image below illustrates the H7N9 virus found in China in 2013.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1007%2Fs11908-018-0642-9/MediaObjects/11908_2018_642_Fig2_HTML.png?as=webp" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Goneau et al.&nbsp;2018</figcaption>
</figure>
</div>
<p>Below illustrates the re-assortments for some of the pandemics.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="https://media.springernature.com/full/springer-static/image/art%3A10.1007%2Fs11908-018-0642-9/MediaObjects/11908_2018_642_Fig3_HTML.png?as=webp" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Goneau et al.&nbsp;2018</figcaption>
</figure>
</div>
<p>A video from The WHO</p>
<iframe width="775" height="300" src="https://www.youtube.com/embed/mn710cT4xxs" title="What is influenza?" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen="">
</iframe>
<p>A very interesting TED talk</p>
<iframe width="775" height="300" src="https://www.youtube.com/embed/nncPtxLCPrE" title="Seth Berkley: HIV and flu -- the vaccine strategy" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen="">
</iframe>
</section>
<section id="practical" class="level2">
<h2 class="anchored" data-anchor-id="practical">Practical</h2>
<p>Background: In 2013 a novel influenza A virus which infects humans was identified in China. Liu et al (2013) published the work in The Lancet.</p>
<p><a href="https://www.sciencedirect.com/science/article/abs/pii/S0140673613609381">Paper</a></p>
<p>We will do something similar, but more obvious. Looking at the tables above showing the hemagglutinin and neuraminidase we see for example that H1 and N1 can be found in humans, birds and pigs. Now imagine genetic drift in these segments in their respective hosts. These segments could have diverged from the ‘wild type’ quite significantly in order to escape immune pressure. Let’s say we now get intra-subtype re-assortment, these segments can be so different that they can evade vaccine-acquired immunity and cause a pandemic.</p>
</section>
<section id="methods" class="level2">
<h2 class="anchored" data-anchor-id="methods">Methods:</h2>
<p>Disclaimer: It is important to note that this exercise is completely fictional and its purpose is to illustrate a concept.</p>
<p>Data:</p>
<p><a href="data/HA_85_nr.faa">HA</a></p>
<p><a href="data/NA_85_nr.faa">NA</a></p>
<p>We will use this site for our analysis: <a href="https://www.phylogeny.fr/alacarte.cgi">Phylogeny.fr</a></p>
<p>or newer <a href="https://ngphylogeny.fr/workflows/alacarte">https://ngphylogeny.fr/workflows/alacarte</a></p>
<p>And we will use the ‘A la Carte’ mode and infer our unknown respiratory swab to infer a subtype of H and N.</p>
<p>The easiest way to learn is to do.</p>
<p>Next we will interpret the results to try and figure out the reassortment. Remember this example is contrived, but illustrates a use case for phylogenetic analysis.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>